gef-ui/server/controllers/application-details/index.js

```diff
@@ -347,10 +346,68 @@ const applicationDetails = async (req, res, next) => {
       params.link += '/unissued-facilities';
     }
 
-    const amendmentsInProgressOnDeal = await getPortalAmendmentsOnDeal({ dealId, statuses: PORTAL_AMENDMENT_INPROGRESS_STATUSES, userToken });
+    const amendmentsUnderwayOnDeal = await getPortalAmendmentsOnDeal({ dealId, statuses: PORTAL_AMENDMENT_INPROGRESS_STATUSES, userToken });
 
-    params.canIssuedFacilitiesBeAmended =
-      canUserAmendIssuedFacilities(application.submissionType, application.status, userRoles) && !amendmentsInProgressOnDeal.length;
+    // array of facility ids that are currently being amended
+    const amendmentsInProgress = amendmentsUnderwayOnDeal.map((amendment) => ({
+      amendmentId: amendment.amendmentId,
+      facilityId: amendment.facilityId,
+      status: amendment.status,
+    }));
+
+    params.readyForCheckerAmendmentStatusHeading = PORTAL_AMENDMENT_STATUS.READY_FOR_CHECKERS_APPROVAL;
+    params.readyForCheckerAmendmentDetailsUrlAndText = [];
+
+    params.furtherMakersInputAmendmentStatusHeading = PORTAL_AMENDMENT_STATUS.FURTHER_MAKERS_INPUT_REQUIRED;
+    params.furtherMakersInputAmendmentDetailsUrlAndText = [];
+
+    /**
+     * Flag facilities that are issued and have an amendment in progress
+     * maps through facilities
+     * checks if facility is issued, checks if facility does not have an amendment in progress and checks if user can amend issued facilities
+     * if all 3 are true, set canIssuedFacilitiesBeAmended to true
+     */
+    const facilitiesWithAmendmentFlag = params.facilities.data.map((eachFacility) => {
```

Comment:
```suggestion
    const facilitiesWithAmendmentFlag = params.facilities.data.map((facility) => {
```

Reply:
This function should be a helper function given the complexity and re-usability of the logic.